src/lib/y2storage/proposal/space_maker.rb

```diff
@@ -306,95 +291,89 @@ def force_ptables(planned_partitions)
       # @param planned_partitions [Array<Planned::Partition>] set of partitions to make space for
-      # @param prospects [SpaceMakerProspects::List] set of prospect actions
-      #   that could be executed
       # @param disk_name [String] optional disk name to restrict operations to
       # @return [Boolean] true if some operation was performed. False if nothing
       #   else could be done to reach the goal.
-      def execute_next_action(planned_partitions, prospects, disk_name = nil)
-        prospect_action = prospects.next_available_prospect
-        if !prospect_action
-          log.info "No more prospects for SpaceMaker (disk_name: #{disk_name})"
+      def execute_next_action(actions, planned_partitions, disk_name = nil)
+        action = actions.next
+        if !action
+          log.info "No more actions for SpaceMaker (disk_name: #{disk_name})"
           return false
         end
 
-        case prospect_action.to_sym
-        when :resize_partition
-          execute_resize(prospect_action, planned_partitions, disk_name)
-        when :delete_partition
-          execute_delete(prospect_action, prospects)
-        else
-          execute_wipe(prospect_action)
-        end
-
+        sids = execute_action(action, new_graph, planned_partitions, disk_name)
+        actions.done(sids)
+        new_graph_deleted_sids.concat(sids)
         true
       end
 
-      # Performs the action described by a {SpaceMakerProspects::ResizePartition}
+      # Executes the given SpaceMaker action
       #
-      # @param prospect [SpaceMakerProspects::ResizePartition] candidate action
-      #   to execute
-      # @param planned_partitions [Array<Planned::Partition>] set of partitions to make space for
+      # @param action [SpaceMakerActions::Base] action to execute
+      # @param graph [Devicegraph] devicegraph in which the action will be executed
+      # @param planned_partitions [Array<Planned::Partition>, nil] set of partitions to make space
+      #   for, if known. Nil for mandatory operations executed during {#prepare_devicegraph}.
       # @param disk_name [String, nil] optional disk name to restrict operations to
-      def execute_resize(prospect, planned_partitions, disk_name)
-        log.info "SpaceMaker#execute_resize - #{prospect}"
-
-        part = new_graph.find_device(prospect.sid)
-        target_shrink_size = resizing_size(part, planned_partitions, disk_name)
-        shrink_partition(part, target_shrink_size)
-        prospect.available = false
+      # @return [Array<Integer>] sids of the devices that has been deleted as a consequence
+      #   of the action
+      def execute_action(action, graph, planned_partitions = nil, disk_name = nil)
+        case action
+        when SpaceMakerActions::Shrink
+          execute_shrink(action, graph, planned_partitions, disk_name)
+        when SpaceMakerActions::Delete
+          execute_delete(action, graph)
+        else
```

Comment:
NP: I would explicitly check by `SpaceMakerActions::Shrink`, and loging/raising an exception in else.

Reply:
I fear that would only lead to a line of code not covered by tests. This is quite internal, not something receiving direct input from the caller. I wouldn't expect a `SpaceMakerActions::List` to return an action from an unknown type.